### ISSUE TITLE
[Calendar] Resetting minDate and maxDate to null doesn't re-enable full date range in popup

### ISSUE THREAD

**Issue by h0lg** [fomantic/Fomantic-UI#1776]
# Bug Report
Resetting minDate and maxDate to null doesn't re-enable the full date range in the popup.

## Steps to reproduce
1. Initialize a calendar
2. Set its minDate and maxDate
3. Reset its minDate and maxDate to null
4. Open the calendar widget

## Expected result
The full date range should be enabled in the popup.

## Actual result
Only the date range set in step 2 is enabled.

## Testcase
```js
    //just a helper
    Date.prototype.addDays = function(days) {
        var date = new Date(this.valueOf());
        date.setDate(date.getDate() + days);
        return date;
    };

    var today = new Date(new Date().toDateString()),
        $widget = $('#standard_calendar').calendar({ type: 'date' });
        
    $widget.calendar('set minDate', today.addDays(-1));
    $widget.calendar('set maxDate', today.addDays(1));
    $widget.calendar('set minDate', null);
    $widget.calendar('get minDate'); //correctly returns null
    $widget.calendar('set maxDate', null);
    $widget.calendar('get maxDate'); //correctly returns null    
    $widget.calendar('refresh'); //doesn't help either
    $widget.calendar('popup', 'show'); //still only enables the date range set in step 2
```
[Play with my fiddle.](https://jsfiddle.net/cr7myt25/2/)

## Version
2.8.7

**Comment by lubber-de** [comment]
Fixed by #1777 See your adjusted jsfidde here
https://jsfiddle.net/ko2in/8ym30fju/8/

**Comment by h0lg** [comment]
Looks good, thanks for the quick response! Looking forward to 2.8.8 .
